test/test.js

```diff
@@ -893,6 +893,57 @@ describe('sanitizeHtml', function() {
       }), '<span style="color:yellow;text-align:center;font-family:helvetica"></span>'
     );
   });
+  it('should delete the script tag', function() {
+    assert.equal(sanitizeHtml('<script src="https://www.unauthorized.com/lib.js"></script>', {
+      allowedTags: [ 'script' ],
+      allowVulnerableTags: true,
+      allowedAttributes: {
+        script: [ 'src' ]
+      },
+      allowedScriptHostnames: [ 'www.authorized.com' ]
+    }), '<script></script>');
+  });
+  it('should delete the script tag since src is not a valid URL', function() {
+    assert.equal(sanitizeHtml('<script src="not-a-valid-url"></script>', {
+      allowedTags: [ 'script' ],
+      allowVulnerableTags: true,
+      allowedAttributes: {
+        script: [ 'src' ]
+      },
+      allowedScriptHostnames: [ 'www.unauthorized.com' ]
+    }), '<script></script>');
+  });
+  it('Should allow domains in a script that are whitelisted', function() {
+    assert.equal(
+      sanitizeHtml('<script src="https://www.safe.authorized.com/lib.js"></script>', {
+        allowedTags: [ 'script' ],
+        allowedAttributes: {
+          script: [ 'src' ]
+        },
+        allowedScriptDomains: [ 'authorized.com' ]
+      }), '<script src="https://www.safe.authorized.com/lib.js"></script>'
+    );
+  });
+  it('should delete the script tag content', function() {
+    assert.equal(sanitizeHtml('<script src="https://www.authorized.com/lib.js"> alert("evil") </script>', {
+      allowedTags: [ 'script' ],
+      allowVulnerableTags: true,
+      allowedAttributes: {
+        script: [ 'src' ]
+      },
+      allowedScriptHostnames: [ 'www.authorized.com' ]
+    }), '<script src="https://www.authorized.com/lib.js"></script>');
+  });
+  it('should leave the content of script elements', function() {
```

Comment:
I am not sure what this is testing, since the script element has no content and we would not want to keep it. I think you're testing that it works *without* content, but please clarify what the test is for.

Reply:
It's testing that it allows the tag to be present in the DOM, since the hostname is in the list, I renamed the test, hopefully it's clearer now